$OpenBSD$
- Correct a bug in kk_os_ensure_dir() [still wrong]
- Don't display #pragma for OpenBSD.

Index: kklib/src/os.c
--- kklib/src/os.c.orig
+++ kklib/src/os.c
@@ -303,9 +303,9 @@ kk_decl_export int kk_os_ensure_dir(kk_string_t path, 
       // if not, create each part of the path
       do {
         char c = (char)(*p);
-        if ((c == 0 && p != cpath) || c == '/' || c == '\\') {
+        if ((c == 0 || c == '/' || c == '\\') && (p != cpath)) {
           *p = 0;
           if (!kk_is_dir(cpath)) {
             #if defined(WIN32)
             int res = _wmkdir(cpath);
             #else 
@@ -990,7 +991,7 @@ kk_string_t kk_os_app_path(kk_context_t* ctx) {
 }
 
 #else
-#if !defined(__wasi__) && !defined(__EMSCRIPTEN__)
+#if !defined(__wasi__) && !defined(__EMSCRIPTEN__) && !defined(__OpenBSD__)
 #pragma message("using generic application path detection")
 #endif
 kk_string_t kk_os_app_path(kk_context_t* ctx) {
@@ -1245,4 +1246,4 @@ bool kk_cpu_is_little_endian(kk_context_t* ctx) {
   #else
   return false;
   #endif
-}
\ No newline at end of file
+}
